cmd/thriftbreak/main.go

```diff
@@ -57,8 +59,18 @@ func run(args []string) error {
 	}
 
 	lints := pass.Lints()
+	var out string
 	for _, l := range lints {
-		fmt.Println(l.String())
+		if *jsonOut {
+			bytes, err := json.Marshal(l)
+			if err != nil {
+				return fmt.Errorf("failed to marshal error: %v", err)
+			}
+			out = string(bytes)
+		} else {
```

Comment:
This if-else over here is a bit smelly.

Consider resolving the way output is processed once outside the loop (possibly into a `func(Lint) error`)
and calling that from within the loop.

So maybe something like the following would work:

```go
func readableOutput(w io.Writer) func(Lint) error
func jsonOutput(w io.Writer) func(Lint) error
```

Usage:

```go
var write func(Lint) error
if *json { write = jsonOutput(os.Stdout) }
else { write = readableOutput(os.Stdout) }

for _, l := range lints {
  if err := write(l); err != nil { .. }
}
```

---

For the JSON output, if you use [json.Encoder](https://pkg.go.dev/encoding/json#Encoder), the newline will be included for you:

```
  enc := json.NewEncoder(w)
  return func(l Lint) error {
    // Encode adds a trailing newline.
    if err := enc.Encode(l); err != nil {
        return fmt.Errorf(..)
    }
    return nil
  }
```

---

Lastly, I'm using `func(Lint) error` as the output consumer,
but I'll leave it up to you if that's okay,
or if you'd prefer something like a `Sink interface{ WriteLint(Lint) error }` type.